evaluator/evaluator/tasks/predict_task.py

```diff
@@ -88,42 +88,48 @@ def do_predict(
     headers = {"Authorization": f"Bearer {token}"}
     predictions: List[Prediction] = []
     start_time = datetime.datetime.now()
-    num_samples = 8  # len(dataset)
-    for i in range(num_samples):
-        logger.debug(f"Predicting sample {i + 1}/{num_samples}")
-        reference_data = dataset[i]
-
-        if dataset_metadata.skill_type == "extractive-qa":
-            query_request = {
-                "query": reference_data["question"],
-                "skill_args": {"context": reference_data["context"]},
-                "num_results": 1,
-            }
-        elif dataset_metadata.skill_type == "multiple-choice":
-            query_request = {
-                "query": reference_data["question"],
-                "skill_args": {"choices": reference_data["choices"]},
-                "num_results": 1,
-            }
-        else:
-            skill_type = dataset_metadata.skill_type
-            raise ValueError(
-                f"Predictions on '{skill_type}' datasets are currently not supported."
-            )
 
-        response = requests.post(
-            f"{base_url}/skill-manager/skill/{skill_id}/query",
-            headers=headers,
-            data=json.dumps(query_request),
+    if dataset_metadata["skill-type"] == "extractive-qa":
+        # 62c1ae1b536b1bb18ff91ce3 # squad
+        qa_type = "context"
+    elif dataset_metadata["skill-type"] == "multiple-choice":
+        # 62c1ae19536b1bb18ff91cde # commonsense_qa
+        qa_type = "choices"
+    else:
+        skill_type = dataset_metadata["skill-type"]
+        raise HTTPException(
+            400,
+            f"Predictions on '{skill_type}' datasets is currently not supported.",
         )
-        response.raise_for_status()
-        prediction_response = response.json()["predictions"][0]
 
+    query_request = {
+        "query": [
+            datapoint["question"] for datapoint in dataset if "question" in datapoint
+        ],
+        "skill_args": {
+            "context": [
+                datapoint[qa_type] for datapoint in dataset if qa_type in datapoint
+            ]
+        },
+        "num_results": 1,
+    }
+
+    response = requests.post(
```

Comment:
Wenn ich das richtig verstehe machen wir jetzt alles in 1 Batch?   
Hätte gedacht wir stückeln das in 100er/1000er Batches oder so. Kann ja durchaus Datasets geben die sehr groß sind.

Aber wenn es problemlos geht so... von mir aus :D

Reply:
Ich konnte es ehrlich gesagt nicht mit einem ganzen dataset testen.. Habe nur gemockte Daten für eine query mit batch size = 2, kann aber gerne eine variable batch size setzen und nicht das ganze dataset reinsetzen, Glaube haben darüber sogar schon mal gesprochen, falls ich mich richtig erinnere meinten wir, dass das kein Problem sein sollte.

Reply:
Habs jetzt mit 1000 getestet mal. Das ging. Allerdings gibt das model ja auch sofort eine response (weil gemockt).

Theoretisch sollte das vom Request her gehen (von wegen Größe und so). Einzige Sache ist "wie lange braucht der Skill bis er eine Response gibt?". Nicht, dass es dann einen Timeout gibt beim Request weil es zu lange dauert (nicht weil es einen Fehler gäbe, sondern weil das model halt noch am predicten ist).

Reply:
Wie besprochen mach eigenem Ermessen.

Reply:
Hast du ein Vorschlag, wie ich am besten lösen sollte? In einer Schleife x questions in einem request setzen, bis alle questions abgearbeitet wurden? 
Es ist halt schwierig zu beurteilen, wie lange die skills brauchen bis die predictions berechnet wurden.

Reply:
Ja so hätte ich es jetzt einfach in etwa gemacht. 
Von wegen (Pseudocode, kein Gewähr darauf dass das so klappt xD):
```python
batch_size = 1000 # (batch-size wäre dann am besten eine env Variable)
for i in range(0, len(dataset), batch_size):
    batch = dataset[i:i+batch_size]
    # .... request for alle samples in der batch machen
```

Reply:
Also im Prinzip einfach mehrere Requests hintereinander